src/test/PolygonFraxE2E.t.sol

```diff
@@ -0,0 +1,308 @@
+      variableDebtToken: address(0), // Mock, as they don't get validated, because of the "dynamic" deployment on proposal execution
+      stableDebtToken: address(0), // Mock, as they don't get validated, because of the "dynamic" deployment on proposal execution
+      decimals: 18,
+      ltv: 7500,
+      liquidationThreshold: 8000,
+      liquidationBonus: 10500,
+      liquidationProtocolFee: 1000,
+      reserveFactor: 1000,
+      usageAsCollateralEnabled: true,
+      borrowingEnabled: true,
+      interestRateStrategy: AaveV3Helpers
+        ._findReserveConfig(allConfigsAfter, 'USDT', false)
+        .interestRateStrategy,
+      stableBorrowRateEnabled: false,
+      isActive: true,
+      isFrozen: false,
+      isSiloed: false,
+      supplyCap: 50_000_000,
+      borrowCap: 0,
+      debtCeiling: 2_000_000_00,
+      eModeCategory: 1
+    });
+
+    AaveV3Helpers._validateReserveConfig(expectedAssetConfig, allConfigsAfter);
+
+    AaveV3Helpers._noReservesConfigsChangesApartNewListings(
+      allConfigsBefore,
+      allConfigsAfter
+    );
+
+    AaveV3Helpers._validateReserveTokensImpls(
+      vm,
+      AaveV3Helpers._findReserveConfig(allConfigsAfter, 'FRAX', false),
+      ReserveTokens({
+        aToken: fraxPayload.ATOKEN_IMPL(),
+        stableDebtToken: fraxPayload.SDTOKEN_IMPL(),
+        variableDebtToken: fraxPayload.VDTOKEN_IMPL()
+      })
+    );
+
+    AaveV3Helpers._validateAssetSourceOnOracle(FRAX, fraxPayload.PRICE_FEED());
+
+    // impl should be same as USDC
+    AaveV3Helpers._validateReserveTokensImpls(
+      vm,
+      AaveV3Helpers._findReserveConfig(allConfigsAfter, 'USDC', false),
+      ReserveTokens({
+        aToken: fraxPayload.ATOKEN_IMPL(),
+        stableDebtToken: fraxPayload.SDTOKEN_IMPL(),
+        variableDebtToken: fraxPayload.VDTOKEN_IMPL()
+      })
+    );
+
+    _validatePoolActionsPostListing(allConfigsAfter);
+  }
+
+  function _validatePoolActionsPostListing(
+    ReserveConfig[] memory allReservesConfigs
+  ) internal {
+    address aFRAX = AaveV3Helpers
+      ._findReserveConfig(allReservesConfigs, 'FRAX', false)
+      .aToken;
+    address vFRAX = AaveV3Helpers
+      ._findReserveConfig(allReservesConfigs, 'FRAX', false)
+      .variableDebtToken;
+    address sFRAX = AaveV3Helpers
+      ._findReserveConfig(allReservesConfigs, 'FRAX', false)
+      .stableDebtToken;
+    address vDAI = AaveV3Helpers
+      ._findReserveConfig(allReservesConfigs, 'DAI', false)
+      .variableDebtToken;
+
+    AaveV3Helpers._deposit(
+      vm,
+      FRAX_WHALE,
+      FRAX_WHALE,
+      FRAX,
+      666 ether,
+      true,
+      aFRAX
+    );
+
+    AaveV3Helpers._borrow(vm, FRAX_WHALE, FRAX_WHALE, DAI, 2 ether, 2, vDAI);
+
+    AaveV3Helpers._borrow(
+      vm,
+      FRAX_WHALE,
+      FRAX_WHALE,
+      FRAX,
+      200 ether,
+      2,
+      vFRAX
+    );
+
+    // TODO: uncomment when foundry fixes bug with public methods on libs
```

Comment:
Any update on this?

Reply:
it still not solved looks like. this is the issue that mentions it: https://github.com/foundry-rs/foundry/issues/2549